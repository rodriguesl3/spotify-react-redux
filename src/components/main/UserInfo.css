.user-content {
  color: #FFFFFF;
}

.user-content .card-player {
  color: #001F54;
  border-radius: 25px;
  width: 100%;
  background: white;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

/*

    background: white;
    border-radius: 25px;
    color: black;
*/
/*# sourceMappingURL=UserInfo.css.map */