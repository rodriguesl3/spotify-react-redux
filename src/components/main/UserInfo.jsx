import React from 'react'
import PropTypes from 'prop-types'


export const UserInfo = (props) => {
  const { spotifyUserInfo } = props;

  // const getUserInfoHandle = () => {

  // }


  return (
    <div>
      Welcome
      spotify User Information
      {!spotifyUserInfo && <div>Loading</div>}
      {/* {spotifyUserInfo && JSON.stringify(spotifyUserInfo)} */}


    </div>
  )
}

UserInfo.propTypes = {
  //userName: React.PropTypes.string,
  spotifyUserInfo: PropTypes.object,
}
