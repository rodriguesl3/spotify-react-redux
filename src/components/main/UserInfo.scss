@import '../../base.scss';

.user-content {
    color: $component-color;
    .card-player {
        color:$base-color;
    border-radius: 25px;
    width: 100%;
    background: white;
    display: flex;
    align-items: center;
    justify-content: center;
    }
   

}



/*

    background: white;
    border-radius: 25px;
    color: black;
*/