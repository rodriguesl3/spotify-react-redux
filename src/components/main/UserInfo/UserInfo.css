.user-content {
  color: #FFFFFF;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.user-content h3 {
  margin-bottom: 0;
}

.user-content h2 {
  margin-bottom: 2rem;
}
/*# sourceMappingURL=UserInfo.css.map */