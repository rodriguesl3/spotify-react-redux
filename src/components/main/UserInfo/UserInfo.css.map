{
    "version": 3,
    "mappings": "AAEA,AAAA,aAAa,CAAC;EACV,KAAK,ECQQ,OAAO;EDPpB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,MAAM;EACvB,WAAW,EAAE,MAAM;CAStB;;AAdD,AAOI,aAPS,CAOT,EAAE,CAAC;EACC,aAAa,EAAE,CAAC;CACnB;;AATL,AAWI,aAXS,CAWT,EAAE,CAAC;EACC,aAAa,EAAE,IAAI;CACtB",
    "sources": [
        "UserInfo.scss",
        "../../../base.scss"
    ],
    "names": [],
    "file": "UserInfo.css"
}