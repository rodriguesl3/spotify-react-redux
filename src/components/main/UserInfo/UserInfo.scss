@import '../../../base.scss';

.user-content {
    color: $component-color;

    
}
