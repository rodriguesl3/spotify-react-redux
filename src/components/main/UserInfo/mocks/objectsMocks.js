export const followingArtists = {
    "data": {
        "artists": {
            "items": [{
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/04gDigrS5kc9YWfZHwBETP"
                    },
                    "followers": {
                        "href": null,
                        "total": 15860784
                    },
                    "genres": [
                        "pop"
                    ],
                    "href": "https://api.spotify.com/v1/artists/04gDigrS5kc9YWfZHwBETP",
                    "id": "04gDigrS5kc9YWfZHwBETP",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/4079ef5effe54ab8cc099e2b6d872899d033b09d",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/a6b50b62941f169aea7a48039a509f643c8c69e3",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/34760086fe6f4e6f8121fe00aa4901ace216b268",
                            "width": 160
                        }
                    ],
                    "name": "Maroon 5",
                    "popularity": 90,
                    "type": "artist",
                    "uri": "spotify:artist:04gDigrS5kc9YWfZHwBETP"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/0QaSiI5TLA4N7mcsdxShDO"
                    },
                    "followers": {
                        "href": null,
                        "total": 160854
                    },
                    "genres": [
                        "drum and bass",
                        "edm",
                        "house",
                        "liquid funk"
                    ],
                    "href": "https://api.spotify.com/v1/artists/0QaSiI5TLA4N7mcsdxShDO",
                    "id": "0QaSiI5TLA4N7mcsdxShDO",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/82ce67164bb0cbebb335b71d1619c29658fb2c50",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/ebaabbd21b854b86c5a213b91b1aeef90b2ffb5d",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/37be223ef9ad6188bc40f83968f9b468f131cc44",
                            "width": 160
                        }
                    ],
                    "name": "Sub Focus",
                    "popularity": 62,
                    "type": "artist",
                    "uri": "spotify:artist:0QaSiI5TLA4N7mcsdxShDO"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/0jnsk9HBra6NMjO2oANoPY"
                    },
                    "followers": {
                        "href": null,
                        "total": 6715984
                    },
                    "genres": [
                        "dance pop",
                        "edm",
                        "miami hip hop",
                        "pop",
                        "pop rap"
                    ],
                    "href": "https://api.spotify.com/v1/artists/0jnsk9HBra6NMjO2oANoPY",
                    "id": "0jnsk9HBra6NMjO2oANoPY",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/2d0b5f483c7ed95b7c3c63147c3d38f9366b2c21",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/98f6070f467c5b95e09505aeda1aa5b48d1a133d",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/d1a78628d48a855d51ab7d74b6d78c22ee8a2b63",
                            "width": 160
                        }
                    ],
                    "name": "Flo Rida",
                    "popularity": 81,
                    "type": "artist",
                    "uri": "spotify:artist:0jnsk9HBra6NMjO2oANoPY"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/1Cs0zKBU1kc0i8ypK3B9ai"
                    },
                    "followers": {
                        "href": null,
                        "total": 17501914
                    },
                    "genres": [
                        "dance pop",
                        "edm",
                        "pop"
                    ],
                    "href": "https://api.spotify.com/v1/artists/1Cs0zKBU1kc0i8ypK3B9ai",
                    "id": "1Cs0zKBU1kc0i8ypK3B9ai",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/ce083ade33c047d3f2a72d1cfba3bb2f10e7ecea",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/86277c32db4a89ff74e9478a852e701975a4a49c",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/0de6be7a3a2d1a82a7adafdbb44282b136741150",
                            "width": 160
                        }
                    ],
                    "name": "David Guetta",
                    "popularity": 90,
                    "type": "artist",
                    "uri": "spotify:artist:1Cs0zKBU1kc0i8ypK3B9ai"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/3eH2apcLhzhnr3eWmH3VBf"
                    },
                    "followers": {
                        "href": null,
                        "total": 82998
                    },
                    "genres": [
                        "brazilian electronica",
                        "complextro",
                        "electro house",
                        "melbourne bounce"
                    ],
                    "href": "https://api.spotify.com/v1/artists/3eH2apcLhzhnr3eWmH3VBf",
                    "id": "3eH2apcLhzhnr3eWmH3VBf",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/7e7ef448a57858dc5b580698756aa99d4ec01e59",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/485d6876a9c02d5d218b030ce91f2e4c19da0b22",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/ca633b904c302834b610674cc6124993748270b2",
                            "width": 160
                        }
                    ],
                    "name": "Felguk",
                    "popularity": 63,
                    "type": "artist",
                    "uri": "spotify:artist:3eH2apcLhzhnr3eWmH3VBf"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/4OBJLual30L7gRl5UkeRcT"
                    },
                    "followers": {
                        "href": null,
                        "total": 2688144
                    },
                    "genres": [
                        "atl hip hop",
                        "dance pop",
                        "dirty south rap",
                        "gangster rap",
                        "hip hop",
                        "pop rap",
                        "rap",
                        "southern hip hop",
                        "trap music"
                    ],
                    "href": "https://api.spotify.com/v1/artists/4OBJLual30L7gRl5UkeRcT",
                    "id": "4OBJLual30L7gRl5UkeRcT",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/eae4a88e33baad4b4ce4605aa074c45766e94d09",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/b83211485b5ad5e57b9d3659a096ffef3d6f3171",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/c1cb65dc810e054da565baa5cbb746b6b44a9aed",
                            "width": 160
                        }
                    ],
                    "name": "T.I.",
                    "popularity": 80,
                    "type": "artist",
                    "uri": "spotify:artist:4OBJLual30L7gRl5UkeRcT"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/5WgRuO0mhM36NFoapzpWBH"
                    },
                    "followers": {
                        "href": null,
                        "total": 422166
                    },
                    "genres": [
                        "brazilian hip hop",
                        "brazilian rock"
                    ],
                    "href": "https://api.spotify.com/v1/artists/5WgRuO0mhM36NFoapzpWBH",
                    "id": "5WgRuO0mhM36NFoapzpWBH",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/4d6a194949e2e44568698816bfe8754165415cf5",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/ada70f6d6fb639cf1920a9fdd553a5daecfd9775",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/eccbebafd40a2c5290bb54f8cabb850c7b6da17d",
                            "width": 160
                        }
                    ],
                    "name": "Rashid",
                    "popularity": 62,
                    "type": "artist",
                    "uri": "spotify:artist:5WgRuO0mhM36NFoapzpWBH"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/5X4LWwbUFNzPkEas04uU82"
                    },
                    "followers": {
                        "href": null,
                        "total": 807506
                    },
                    "genres": [
                        "big room",
                        "dance pop",
                        "deep big room",
                        "edm",
                        "electro house",
                        "pop",
                        "progressive electro house",
                        "tropical house"
                    ],
                    "href": "https://api.spotify.com/v1/artists/5X4LWwbUFNzPkEas04uU82",
                    "id": "5X4LWwbUFNzPkEas04uU82",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/6264c2e9be1c901ac1eeb1be3a768097f9bc8f68",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/6f02efa253c9f13409fe8ab6792eca5e2348250e",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/a26ecb563eee666d09c34790f1e0f5ca033c6311",
                            "width": 160
                        }
                    ],
                    "name": "DVBBS",
                    "popularity": 68,
                    "type": "artist",
                    "uri": "spotify:artist:5X4LWwbUFNzPkEas04uU82"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/64KEffDW9EtZ1y2vBYgq8T"
                    },
                    "followers": {
                        "href": null,
                        "total": 8435376
                    },
                    "genres": [
                        "brostep",
                        "edm",
                        "egyptian pop",
                        "pop",
                        "progressive electro house"
                    ],
                    "href": "https://api.spotify.com/v1/artists/64KEffDW9EtZ1y2vBYgq8T",
                    "id": "64KEffDW9EtZ1y2vBYgq8T",
                    "images": [{
                            "height": 640,
                            "url": "https://i.scdn.co/image/65f40b9a99ebb96921d5366537cdff7d5661d914",
                            "width": 640
                        },
                        {
                            "height": 320,
                            "url": "https://i.scdn.co/image/719dab7f27ba0793b4ca7c5672ecf56d3d88467b",
                            "width": 320
                        },
                        {
                            "height": 160,
                            "url": "https://i.scdn.co/image/d390ea8d34e3066c927eca1a67a9e6ccb6e943a1",
                            "width": 160
                        }
                    ],
                    "name": "Marshmello",
                    "popularity": 90,
                    "type": "artist",
                    "uri": "spotify:artist:64KEffDW9EtZ1y2vBYgq8T"
                }
            ],
            "next": null,
            "total": 9,
            "cursors": {
                "after": null
            },
            "limit": 30,
            "href": "https://api.spotify.com/v1/me/following?type=artist&limit=30"
        }
    },
    "status": 200,
    "statusText": "",
    "headers": {
        "content-type": "application/json; charset=utf-8",
        "cache-control": "private, max-age=0"
    },
    "config": {
        "transformRequest": {},
        "transformResponse": {},
        "timeout": 0,
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
        "maxContentLength": -1,
        "headers": {
            "Accept": "application/json, text/plain, */*",
            "Authorization": "Bearer BQAb_nXk7jka9cJbg_53Q_F2rBLOI9E8JUAT1nL1U6A2vAHgHemqhq-UPW8XCXErvoXneKLNtymn0G1_aALl3nvIqGEDqeudG4d6Wc5a0VYB47h_cFVutVN5K_N1YFigIPB7k7AGzTxWHeZlot8_B5yieM3GE7Puz9zDSIxcd_d4en3_ryIEHQIu_HUWz_M"
        },
        "method": "get",
        "url": "https://api.spotify.com/v1/me/following?type=artist&limit=30"
    },
    "request": {}
};

export const userInfo = {
    "data": {
        "country": "BR",
        "display_name": "Lucas Rodrigues",
        "email": "lusk.rs@hotmail.com",
        "external_urls": {
            "spotify": "https://open.spotify.com/user/21aq556qn5jutmhf2jmvqedxa"
        },
        "followers": {
            "href": null,
            "total": 42
        },
        "href": "https://api.spotify.com/v1/users/21aq556qn5jutmhf2jmvqedxa",
        "id": "21aq556qn5jutmhf2jmvqedxa",
        "images": [{
            "height": null,
            "url": "https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=1084373748241702&height=200&width=200&ext=1551600776&hash=AeTTgzEoEZf2Z_uj",
            "width": null
        }],
        "product": "open",
        "type": "user",
        "uri": "spotify:user:21aq556qn5jutmhf2jmvqedxa"
    },
    "status": 200,
    "statusText": "",
    "headers": {
        "content-type": "application/json; charset=utf-8",
        "cache-control": "private, max-age=0"
    },
    "config": {
        "transformRequest": {},
        "transformResponse": {},
        "timeout": 0,
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
        "maxContentLength": -1,
        "headers": {
            "Accept": "application/json, text/plain, */*",
            "Authorization": "Bearer BQAb_nXk7jka9cJbg_53Q_F2rBLOI9E8JUAT1nL1U6A2vAHgHemqhq-UPW8XCXErvoXneKLNtymn0G1_aALl3nvIqGEDqeudG4d6Wc5a0VYB47h_cFVutVN5K_N1YFigIPB7k7AGzTxWHeZlot8_B5yieM3GE7Puz9zDSIxcd_d4en3_ryIEHQIu_HUWz_M"
        },
        "method": "get",
        "url": "https://api.spotify.com/v1/me"
    },
    "request": {}
}

export const userListening = {
    "data": {
        "device": {
            "id": "7fabdf5ffd989d53b3487cdda9c0c547a7e9872a",
            "is_active": true,
            "is_private_session": false,
            "is_restricted": false,
            "name": "LAPTOP-RB4MNQUG",
            "type": "Computer",
            "volume_percent": 55
        },
        "shuffle_state": false,
        "repeat_state": "context",
        "timestamp": 1549011734993,
        "context": {
            "external_urls": {
                "spotify": "https://open.spotify.com/playlist/37i9dQZEVXbtbzuR2SlZM0"
            },
            "href": "https://api.spotify.com/v1/playlists/37i9dQZEVXbtbzuR2SlZM0",
            "type": "playlist",
            "uri": "spotify:user:spotify:playlist:37i9dQZEVXbtbzuR2SlZM0"
        },
        "progress_ms": 264685,
        "item": {
            "album": {
                "album_type": "album",
                "artists": [{
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/4WN5naL3ofxrVBgFpguzKo"
                    },
                    "href": "https://api.spotify.com/v1/artists/4WN5naL3ofxrVBgFpguzKo",
                    "id": "4WN5naL3ofxrVBgFpguzKo",
                    "name": "Rudimental",
                    "type": "artist",
                    "uri": "spotify:artist:4WN5naL3ofxrVBgFpguzKo"
                }],
                "available_markets": [
                    "AD",
                    "AE",
                    "AR",
                    "AT",
                    "AU",
                    "BE",
                    "BG",
                    "BH",
                    "BO",
                    "BR",
                    "CA",
                    "CH",
                    "CL",
                    "CO",
                    "CR",
                    "CY",
                    "CZ",
                    "DE",
                    "DK",
                    "DO",
                    "DZ",
                    "EC",
                    "EE",
                    "EG",
                    "ES",
                    "FI",
                    "FR",
                    "GB",
                    "GR",
                    "GT",
                    "HK",
                    "HN",
                    "HU",
                    "ID",
                    "IE",
                    "IL",
                    "IS",
                    "IT",
                    "JO",
                    "JP",
                    "KW",
                    "LB",
                    "LI",
                    "LT",
                    "LU",
                    "LV",
                    "MA",
                    "MC",
                    "MT",
                    "MX",
                    "MY",
                    "NI",
                    "NL",
                    "NO",
                    "NZ",
                    "OM",
                    "PA",
                    "PE",
                    "PH",
                    "PL",
                    "PS",
                    "PT",
                    "PY",
                    "QA",
                    "RO",
                    "SA",
                    "SE",
                    "SG",
                    "SK",
                    "SV",
                    "TH",
                    "TN",
                    "TR",
                    "TW",
                    "US",
                    "UY",
                    "VN",
                    "ZA"
                ],
                "external_urls": {
                    "spotify": "https://open.spotify.com/album/0ORqx9M6ak7kqDwYUBwvdK"
                },
                "href": "https://api.spotify.com/v1/albums/0ORqx9M6ak7kqDwYUBwvdK",
                "id": "0ORqx9M6ak7kqDwYUBwvdK",
                "images": [{
                        "height": 640,
                        "url": "https://i.scdn.co/image/1a6a17e359b8a39919c541505cd681aef4b7b294",
                        "width": 640
                    },
                    {
                        "height": 300,
                        "url": "https://i.scdn.co/image/d517611a9aef994460eb73e5d38d59149e6f3fef",
                        "width": 300
                    },
                    {
                        "height": 64,
                        "url": "https://i.scdn.co/image/06e2a532cd47b9a3c90f75e281cb382088d197a4",
                        "width": 64
                    }
                ],
                "name": "Toast to our Differences (Deluxe)",
                "release_date": "2019-01-25",
                "release_date_precision": "day",
                "total_tracks": 16,
                "type": "album",
                "uri": "spotify:album:0ORqx9M6ak7kqDwYUBwvdK"
            },
            "artists": [{
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/4WN5naL3ofxrVBgFpguzKo"
                    },
                    "href": "https://api.spotify.com/v1/artists/4WN5naL3ofxrVBgFpguzKo",
                    "id": "4WN5naL3ofxrVBgFpguzKo",
                    "name": "Rudimental",
                    "type": "artist",
                    "uri": "spotify:artist:4WN5naL3ofxrVBgFpguzKo"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/4ScCswdRlyA23odg9thgIO"
                    },
                    "href": "https://api.spotify.com/v1/artists/4ScCswdRlyA23odg9thgIO",
                    "id": "4ScCswdRlyA23odg9thgIO",
                    "name": "Jess Glynne",
                    "type": "artist",
                    "uri": "spotify:artist:4ScCswdRlyA23odg9thgIO"
                },
                {
                    "external_urls": {
                        "spotify": "https://open.spotify.com/artist/7LHLq2wayvWBPTwk6p9zZz"
                    },
                    "href": "https://api.spotify.com/v1/artists/7LHLq2wayvWBPTwk6p9zZz",
                    "id": "7LHLq2wayvWBPTwk6p9zZz",
                    "name": "Chronixx",
                    "type": "artist",
                    "uri": "spotify:artist:7LHLq2wayvWBPTwk6p9zZz"
                }
            ],
            "available_markets": [
                "AD",
                "AE",
                "AR",
                "AT",
                "AU",
                "BE",
                "BG",
                "BH",
                "BO",
                "BR",
                "CA",
                "CH",
                "CL",
                "CO",
                "CR",
                "CY",
                "CZ",
                "DE",
                "DK",
                "DO",
                "DZ",
                "EC",
                "EE",
                "EG",
                "ES",
                "FI",
                "FR",
                "GB",
                "GR",
                "GT",
                "HK",
                "HN",
                "HU",
                "ID",
                "IE",
                "IL",
                "IS",
                "IT",
                "JO",
                "JP",
                "KW",
                "LB",
                "LI",
                "LT",
                "LU",
                "LV",
                "MA",
                "MC",
                "MT",
                "MX",
                "MY",
                "NI",
                "NL",
                "NO",
                "NZ",
                "OM",
                "PA",
                "PE",
                "PH",
                "PL",
                "PS",
                "PT",
                "PY",
                "QA",
                "RO",
                "SA",
                "SE",
                "SG",
                "SK",
                "SV",
                "TH",
                "TN",
                "TR",
                "TW",
                "US",
                "UY",
                "VN",
                "ZA"
            ],
            "disc_number": 1,
            "duration_ms": 268653,
            "explicit": false,
            "external_ids": {
                "isrc": "GBAHS1800369"
            },
            "external_urls": {
                "spotify": "https://open.spotify.com/track/5gHm37FoBAppgJjGI5EA0w"
            },
            "href": "https://api.spotify.com/v1/tracks/5gHm37FoBAppgJjGI5EA0w",
            "id": "5gHm37FoBAppgJjGI5EA0w",
            "is_local": false,
            "name": "Dark Clouds (feat. Jess Glynne & Chronixx)",
            "popularity": 62,
            "preview_url": "https://p.scdn.co/mp3-preview/f40b823251dac30a4e5122e253f93903bd2a1c2c?cid=dc6af13b21094058a9b5e104eccee32e",
            "track_number": 3,
            "type": "track",
            "uri": "spotify:track:5gHm37FoBAppgJjGI5EA0w"
        },
        "currently_playing_type": "track",
        "is_playing": true
    },
    "status": 200,
    "statusText": "",
    "headers": {
        "content-type": "application/json; charset=utf-8",
        "cache-control": "private, max-age=0"
    },
    "config": {
        "transformRequest": {},
        "transformResponse": {},
        "timeout": 0,
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
        "maxContentLength": -1,
        "headers": {
            "Accept": "application/json, text/plain, */*",
            "Authorization": "Bearer BQAb_nXk7jka9cJbg_53Q_F2rBLOI9E8JUAT1nL1U6A2vAHgHemqhq-UPW8XCXErvoXneKLNtymn0G1_aALl3nvIqGEDqeudG4d6Wc5a0VYB47h_cFVutVN5K_N1YFigIPB7k7AGzTxWHeZlot8_B5yieM3GE7Puz9zDSIxcd_d4en3_ryIEHQIu_HUWz_M"
        },
        "method": "get",
        "url": "https://api.spotify.com/v1/me/player"
    },
    "request": {}
}