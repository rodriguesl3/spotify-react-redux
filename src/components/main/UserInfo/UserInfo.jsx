import React from 'react';
import PropTypes from 'prop-types';
import './UserInfo.scss';

import { imageUrl } from '../../../constants/index';
import { CustomCard } from '../../atoms/CustomCard/CustomCard';
import LoaderHOC from '../../atoms/Loading/LoaderHOC';
import UserListen from '../../atoms/UserListen/UserListen';

const UserInfo = (props) => {
  const { spotifyUserInfo, userListening } = props;
  return (
    <div className="user-content">
      <h3>
        Welcome
        {spotifyUserInfo && spotifyUserInfo.display_name}
      </h3>
      <div>
        <h5>That you are listen now?</h5>
        <CustomCard
          image={userListening.item.album ? userListening.item.album.images[1].url : imageUrl}
        >
          <UserListen
            userListening={userListening}
          />
        </CustomCard>
      </div>
    </div>
  );
};

UserInfo.propTypes = {
  userListening: PropTypes.object,
  spotifyUserInfo: PropTypes.object,
};


export default LoaderHOC('userListening', UserInfo);
