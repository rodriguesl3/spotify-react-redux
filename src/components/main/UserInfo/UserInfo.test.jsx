/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from './UserInfo';
import { userInfo, userListening } from './__mocks__/objectsMocks';

test('UserInfo snaphot test', () => {
  const component = renderer.create(
    <UserInfo
      userListening={userListening.data}
      spotifyUserInfo={userInfo.data}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
