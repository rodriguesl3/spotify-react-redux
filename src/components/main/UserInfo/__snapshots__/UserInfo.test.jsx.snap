// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo snaphot test 1`] = `
<div
  className="user-content"
>
  <h3>
    Welcome
    Lucas Rodrigues
  </h3>
  <div>
    <h5>
      That you are listen now?
    </h5>
    <div
      className="col-10 col-sm-4 col-lg-3 custom-card"
    >
      <div
        className="row"
      >
        <div
          className="col-5"
        >
          <img
            alt="card-name"
            className="card-img-top"
            src="https://i.scdn.co/image/d517611a9aef994460eb73e5d38d59149e6f3fef"
          />
        </div>
        <div
          className="col-7"
        >
          <div>
            <span>
              Device: 
              LAPTOP-RB4MNQUG
            </span>
            <p>
              Type: 
              Computer
            </p>
            <div
              className="row"
            >
              <div
                className="col-1"
              >
                <p>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M371.7 238l-176-107c-15.8-8.8-35.7 2.5-35.7 21v208c0 18.4 19.8 29.8 35.7 21l176-101c16.4-9.1 16.4-32.8 0-42zM504 256C504 119 393 8 256 8S8 119 8 256s111 248 248 248 248-111 248-248zm-448 0c0-110.5 89.5-200 200-200s200 89.5 200 200-89.5 200-200 200S56 366.5 56 256z"
                    />
                  </svg>
                </p>
              </div>
              <div
                className="col-9"
              >
                <span>
                  Dark Clouds (feat. Jess Glynne & Chronixx)
                </span>
              </div>
            </div>
            <span>
              <div>
                <div
                  className="stars-outer"
                >
                  <div
                    className="stars-inner"
                    style={
                      Object {
                        "width": "62%",
                      }
                    }
                  />
                </div>
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
